components/sup/src/manager/service_updater.rs

```diff
@@ -197,7 +197,26 @@ impl ServiceUpdater {
                                                                        0);
                                 *st = RollingState::InElection
                             }
-                            _ => return None,
+                            (Some(_), None) => {
+                                debug!("No leader present; rolling Update cannot proceed until \
+                                        the {} group election finishes",
+                                       &service.service_group);
+                                return None;
+                            }
+                            (..) => {
```

Comment:
What's the difference between `_` and `(..)` here?

Reply:
It's the same as `(_,_)` (and `(_,_,_)`, and `(_,_,_,_)`, and...).

Basically, a tuple of some length whose contents are unimportant to you.

Reply:
I see the  same pattern  a little further down in `RollingState::InElection`. I think it might be useful to have the same but slightly different logging there to ensure we have the same visibility in a scenario where an election started but cannot complete satisfactorily for whatever reason. Thoughts?

Reply:
@christophermaier So we've essentially lost the catch-all then?  Why not leave it as `_`, as the original code was?

Reply:
@raskchanky We dont lose anything by changing to `(..)` from `_`. The rust compiler will not let you have non-exhaustive pattern matching. In fact I would prefer `(None, _)` to be more explicit.

Reply:
@davidMcneil That's a good call... I'll switch to `(None, _)` for explicitness.

Reply:
@jeremymv2 Also a good call :+1: Will add!